Dockerfile

```diff
@@ -7,10 +7,14 @@ ARG RUST_IMAGE=ghcr.io/linkerd/dev:v48-rust
 
 # Use an arbitrary ~recent edge release image to get the proxy
 # identity-initializing and linkerd-await wrappers.
+#
+# To use an image that includes a shell, run the following:
+#
+# ```
+# just docker --build-arg LINKERD2_IMAGE='debian:bookworm-slim'
+# ```
```

Comment:
i added this comment in the second commit. i found that running

```
export LINKERD_TAG=$(linkerd version --client --short)
just docker --build-arg LINKERD2_IMAGE='debian:bookworm-slim' && \
  just k3d-create && \
  just k3d-load-linkerd && \
  just linkerd-install
```

...and inspecting `linkerd check`  worked. i'm unsure why that didn't seem to introduce any `linkerd-await`-related failures, but it does appear to provide us with a solution that (a) simplifies this image, (b) brings us into line with production deployments, and (c) offers developers a convenient escape hatch to attach a shell to a running proxy container when needed.

Reply:
after some further digging, i am going to back out of this change. the ephemeral debug container looked really promising, but i found in practice it wasn't yielding me helpful visibility.